src/modules/ztsi/tests/ZtsiTests.cpp

```diff
@@ -75,6 +81,34 @@ namespace OSConfig::Platform::Tests
     Ztsi* ZtsiTests::ztsi;
     std::string ZtsiTests::filename = "./ztsi/config.temp.json";
 
+    TEST_F(ZtsiTests, GetSetServiceUrl)
+    {
+        char serviceUrl[] = "\"https://localhost:8080/\"";
+        MMI_JSON_STRING payload = nullptr;
+        int payloadSizeBytes = 0;
+
+        ASSERT_EQ(MMI_OK, ztsi->Set(g_componentName, g_desiredServiceUrl, serviceUrl, sizeof(serviceUrl)));
+        ASSERT_EQ(MMI_OK, ztsi->Get(g_componentName, g_reportedServiceUrl, &payload, &payloadSizeBytes));
```

Comment:
Add Set and Get tests that use invalid arguments? Null values, unsupported values, etc.

Reply:
Same for next

Reply:
Added here and to Sample CPP module